src/app/questionnaire-item/store/transform-response.ts

```diff
@@ -44,29 +45,38 @@ const getAnswerValueWithQuestionnaireItem = ({
           }
         : undefined;
     case 'date':
-      return {
-        valueDate: isDate(answer)
-          ? toLocaleDate(answer)
-          : isString(answer)
-          ? answer
-          : undefined,
-      };
+      const valueDate = isDate(answer)
+        ? toLocaleDate(answer)
+        : isString(answer)
+        ? answer
+        : undefined;
+      return (
+        valueDate && {
+          valueDate,
+        }
```

Comment:
```suggestion
        valueDate ? {
          valueDate,
        } : undefined,
```

I would prefer a ternary operator as this would make sure that only `undefined` or the object is returned (in the current implementation it could be an empty string as well). It is not crucial in this case as spreading an empty string is the same as spreading `undefined` but I think it would be cleaner nevertheless.

This applies to the dateTime and time below as well.